mesa/time.py

```diff
@@ -264,8 +267,9 @@ def step(self, shuffle_types: bool = True, shuffle_agents: bool = True) -> None:
             shuffle_agents: If True, the order of execution of each agents in a
                             type group is shuffled.
         """
-        type_keys: list[type[Agent]] = list(self.agents_by_type.keys())
+        type_keys: KeysView[type[Agent]] = self.agents_by_type.keys()
         if shuffle_types:
+            type_keys = list(type_keys)
```

Comment:
The type doesn't seem to be consistent here, because it was a `KeysView`, then here it becomes a `list`.

Reply:
Made it Iterable

Reply:
We should enable Mypy in CI so that I don't have to catch this manually.